profiling_router.go

```diff
@@ -0,0 +1,146 @@
+package handlers
+
+import (
+	"context"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"strconv"
+
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/go-utils/logger"
+)
+
+const PprofRoutePrefix = "/debug/pprof"
+
+type profiling struct {
+	Enable bool
+	Auth   authentication
+}
+
+type authentication struct {
+	Username string
+	Password string
+}
+
+func NewProfilingRouter(ctx context.Context) (*Router, error) {
+	log := logger.Get(ctx)
+
+	prof := new(profiling)
+
+	err := prof.initialize()
+	if err != nil {
+		return &Router{}, errgo.Notef(err, "fail to initialize pprof profiling")
+	}
+
+	if !prof.isActivable() {
+		return &Router{}, nil
+	}
+
+	r := NewRouter(log)
+
+	log.Info("Add basic authentication middleware to access profiling routes")
+	r.Use(ErrorMiddleware)
+	r.Use(AuthMiddleware(func(user, password string) bool {
+		return user == prof.Auth.Username && password == prof.Auth.Password
+	}))
+
+	log.Info("Enabling pprof endpoints under " + PprofRoutePrefix)
+
+	r.HandleFunc(PprofRoutePrefix+"/", index)
+	r.HandleFunc(PprofRoutePrefix+"/profile", profile)
+	r.HandleFunc(PprofRoutePrefix+"/symbol", symbol)
+	r.HandleFunc(PprofRoutePrefix+"/cmdline", cmdline)
+	r.HandleFunc(PprofRoutePrefix+"/trace", trace)
+	r.HandleFunc(PprofRoutePrefix+"/allocs", allocs)
+	r.HandleFunc(PprofRoutePrefix+"/heap", heap)
+	r.HandleFunc(PprofRoutePrefix+"/mutex", mutex)
+	r.HandleFunc(PprofRoutePrefix+"/goroutine", goroutine)
+	r.HandleFunc(PprofRoutePrefix+"/threadcreate", threadcreate)
+	r.HandleFunc(PprofRoutePrefix+"/block", block)
+
+	return r, nil
+}
+
+func (prof *profiling) initialize() error {
+	profEnable := os.Getenv("PPROF_ENABLED")
```

Comment:
If the env variable is named `PPROF_ENABLED`, the variable must be named `pprofEnable`

```suggestion
	pprofEnable := os.Getenv("PPROF_ENABLED")
```